mixin/smarty-v2@1/mixin.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * Auto-register "templates/" folder.
+ *
+ * @mixinName smarty-v2
+ * @mixinVersion 1.0.0
+ * @since 5.58
+ *
+ * @param CRM_Extension_MixInfo $mixInfo
+ *   On newer deployments, this will be an instance of MixInfo. On older deployments, Civix may polyfill with a work-a-like.
+ * @param \CRM_Extension_BootCache $bootCache
+ *   On newer deployments, this will be an instance of MixInfo. On older deployments, Civix may polyfill with a work-a-like.
+ */
+return function ($mixInfo, $bootCache) {
+  $dir = $mixInfo->getPath('templates');
+  if (!file_exists($dir)) {
+    return;
+  }
+
+  $register = function() use ($dir) {
+    // This implementation is useful for older versions of CiviCRM. It can be replaced/updated going forward.
+    $smarty = CRM_Core_Smarty::singleton();
```

Comment:
Note the version of this we have been adding to core extensions / civix upgraded extensions is the `addTemplateDir` - which HAS been around for 9 years - so I think we can still switch to that even in the mixin

Reply:
Yes, `addTemplateDir()` exists there, but it doesn't dedupe registrations.

We can theoretically dedupe registrations with some `static` guards (*like in the prior `hook_config` boilerplate*), but TBH my head starts to hurt when I think about whether to use `static $foo` or `Civi::$statics` for this spot. This formulation felt easier to understand. (*I'm really not certain which lifecycle aligns best with the Smarty singleton.*)

For the v1.1.0, it would call the new `addTemplateDir()` which could perhaps do its own deduping?

Reply:
OK - that makes sense